group-coordinator/src/main/java/org/apache/kafka/coordinator/group/modern/SubscribedTopicDescriberImpl.java

```diff
@@ -79,49 +81,65 @@ public int numPartitions(Uuid topicId) {
      */
     @Override
     public Set<String> racksForPartition(Uuid topicId, int partition) {
+        TopicImage topic = metadataImage.topics().getTopic(topicId);
+        if (topic != null) {
+            PartitionRegistration partitionRegistration = topic.partitions().get(partition);
+            if (partitionRegistration != null) {
+                Set<String> racks = new HashSet<>();
+                for (int replica : partitionRegistration.replicas) {
+                    // Only add the rack if it is available for the broker/replica.
+                    metadataImage.cluster().broker(replica).rack().ifPresent(racks::add);
```

Comment:
Should we check the offline replica?

Reply:
I was assuming that we should not. If I recall correctly, this is what we did in KIP-881. I need to double check to be sure.

Reply:
In KIP-881, we don't check whether replicas are online or offline. We basically assume that they will eventually come back. I would follow the same approach here.

Side note, this needs to be aligned with the computation of the hash. There, we include all replicas regardless of whether they are online or not.

Reply:
If the offline replicas could be existent in this path, we need to add null check for `broker`, right?

Reply:
We do null check in hash function. It's better to add the check here.

https://github.com/apache/kafka/blob/ce4940f9891a96819e54f8db097ce3824876e8e5/group-coordinator/src/main/java/org/apache/kafka/coordinator/group/Utils.java#L408-L413